master/internal/api_checkpoint.go

```diff
@@ -28,6 +35,82 @@ func (a *apiServer) GetCheckpoint(
 	}
 }
 
+func (a *apiServer) DeleteCheckpoints(
+	ctx context.Context,
+	req *apiv1.DeleteCheckpointsRequest) (*apiv1.DeleteCheckpointsResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx, a.m.db, &a.m.config.InternalConfig.ExternalSessions)
+	if err != nil {
+		return nil, err
+	}
+
+	checkpointsToDeleteStr := req.CheckpointUuids
+
+	conv := &protoconverter.ProtoConverter{}
+	checkpointsToDelete := conv.ToUUIDList(checkpointsToDeleteStr)
+	if cErr := conv.Error(); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "converting checkpoint: %s", cErr)
+	}
+
+	registeredCheckpointUUIDs, err := a.m.db.FilterForRegisteredCheckpoints(checkpointsToDelete)
+	if err != nil {
+		return nil, err
+	}
+
+	// return 400 if model registry checkpoints and include all the model registry checkpoints
+	if len(registeredCheckpointUUIDs) > 0 {
+		return nil, status.Errorf(codes.InvalidArgument,
+			"This subset of checkpoints provided are in the model registry and cannot be deleted: %v.",
+			registeredCheckpointUUIDs)
+	}
+
+	addr := actor.Addr(fmt.Sprintf("checkpoints-gc-%s", uuid.New().String()))
+
+	taskSpec := *a.m.taskSpec
+	agentUserGroup, err := a.m.db.AgentUserGroup(curUser.ID)
```

Comment:
todo for us: make all the defaulting logic (like agentUserGroup here) way clearer and harder to mess  up